flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessageParser.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.runtime.io.network.NetworkClientHandler;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+
+import java.net.ProtocolException;
+
+/**
+ * Responsible for offering the message header length and parsing the message header part of the message.
+ */
+class NettyMessageParser {
+    /**
+     * Indicates how to deal with the data buffer part of the current message.
+     */
+    public enum DataBufferAction {
```

Comment:
This clarification actually has two dimensions and would bring ambiguous. One dimension is for distinguish between `BufferResponse` and other messages. Another dimension is for distinguish only for `BufferResponse` whether the respective channel is released or not. 

I guess we can refactor the related process to avoid this enum type.

Reply:
Another concern for this enum is that if we add another new type message future, it is hard to know which enum type it belongs to and how it adjust the logics for the new message in below `#parseMessageHeader`.